src/ig/IGExporter.ts

```diff
@@ -444,6 +456,54 @@ export class IGExporter {
     }
   }
 
+  /**
+   * Adds generated pages for each artifact type included in the IG.
+   * Only add pages if there is not already an existing page with the name that would be added.
+   */
+  private addArtifactPages(igPath: string): void {
+    // profiles page has a table with 3 columns:
+    // Profile, Derived From, Description
+    // other pages have tables with 2 columns:
+    // [Type], Description
+    const pageContentExportPath = path.join(igPath, 'input', 'pagecontent');
+    ensureDirSync(pageContentExportPath);
+    for (const artifactType of Object.keys(this.artifacts) as (keyof ArtifactCollection)[]) {
+      const artifactPagePath = path.join(pageContentExportPath, `${artifactType}.md`);
+      if (existsSync(artifactPagePath)) {
+        // overwriting the file is okay only if the existing file looks like it was generated by this function
+        const existingContent = readFileSync(artifactPagePath, 'utf-8');
+        if (!existingContent.startsWith('<!-- Generated by SUSHI -->')) {
+          logger.error(
+            `A page named ${artifactType}.md already exists. No artifact page will be created at this path.`
+          );
+          continue;
+        }
+      }
+      let content = '<!-- Generated by SUSHI -->\n';
+      if (this.artifacts[artifactType].length > 0) {
+        if (artifactType === 'profiles') {
+          content += '| Profiles | Derived From | Description |\n| --- | --- | --- |\n';
+        } else {
+          content += `| ${upperFirst(artifactType)} | Description |\n| --- | --- |\n`;
+        }
+        content += sortBy(this.artifacts[artifactType], 'name')
+          .map(info => {
+            if (artifactType === 'profiles') {
+              return `| [${info.name}](${info.url} "${info.referenceKey}") | [${info.parent}](${
+                info.parentUrl
+              }) | ${info.description ?? ''} |`;
+            } else {
+              return `| [${info.name}](${info.url} "${info.referenceKey}") | ${
+                info.description ?? ''
+              } |`;
+            }
+          })
+          .join('\n');
+        outputFileSync(artifactPagePath, content);
```

Comment:
I don't think there's much we write out anymore besides the FHIR resources, but I was feeling a little weird that we generated files to somewhere besides `fsh-generated`. But I tested moving the files, and the IG publisher doesn't pick up the files if I move them inside `fsh-generated/pagecontent/`, so I think the comment is a very clear indication that it isn't safe to change that file.